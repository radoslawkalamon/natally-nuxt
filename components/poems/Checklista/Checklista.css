.checklista {
  display: flex;
  flex-flow: column nowrap;
  gap: var(--gap);

  --checkbox-size: 20px;
}

.checklista__last-line {
  margin: 0;
  text-align: right;
}

.checklista__gradient {
  background: linear-gradient(to left, transparent 0%, var(--color-base) 100%);
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
}

.checklista__checkbox {
  align-items: center;
  display: flex;
  gap: 12px;
  position: relative;
}

.checklista__input {
  appearance: none;
  left: 0;
  margin: 0;
  position: absolute;
  top: 0;
  visibility: hidden;
}

/** Stale */
.checklista__input + .checklista__fake-input {
  border: var(--border-size) solid var(--color-base);
  border-radius: calc(var(--border-radius) / 2);
  box-sizing: border-box;
  cursor: pointer;
  height: var(--checkbox-size);
  margin: 0;
  min-width: var(--checkbox-size);
  position: relative;
  transition: background-color var(--time-animation) ease, border var(--time-animation) ease;
  width: var(--checkbox-size);
}

.checklista__input + .checklista__fake-input::before,
.checklista__input + .checklista__fake-input::after {
  background-color: var(--color-base);
  border-radius: 1px;
  content: '';
  display: block;
  height: 2px;
  left: 2px;
  opacity: 0;
  position: absolute;
  top: calc(50% - 1px);
  transition: opacity var(--time-animation) ease;
  width: calc(var(--checkbox-size) - 6px);
  z-index: 1;
}

.checklista__input + .checklista__fake-input::before {
  transform: rotate(-45deg);
}

.checklista__input + .checklista__fake-input::after {
  transform: rotate(45deg);
}

/** Checked */
.checklista__input:checked + .checklista__fake-input {
  background-color: var(--color-accent);
  border-color: var(--color-accent);
}

.checklista__input:checked + .checklista__fake-input::before,
.checklista__input:checked + .checklista__fake-input::after {
  background-color: var(--color-off-white-1);
  opacity: 1;
}
