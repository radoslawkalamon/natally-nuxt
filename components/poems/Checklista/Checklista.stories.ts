import type { Meta, Story } from '@storybook/vue'
import PoemsChecklista from '@/components/poems/Checklista/Checklista.vue'
import WrappersTextPoems from '@/components/wrappers/TextPoems/TextPoems.vue'

const meta: Meta = {
  title: 'Poems / Checklista',
  component: PoemsChecklista
}
export default meta

export const Default: Story = () => ({
  components: {
    PoemsChecklista,
    WrappersTextPoems
  },
  template: `<div class="scoped__storybook-section">
    <WrappersTextPoems>
      <PoemsChecklista />
    </WrappersTextPoems>
  </div>`
})
