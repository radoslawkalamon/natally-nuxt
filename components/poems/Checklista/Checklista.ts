import Vue from 'vue'

export default Vue.extend({
  name: 'PoemsChecklista',
  components: {
    ComponentsCheckbox: () => import(/* webpackChunkName: "components-checkbox" */'@/components/components/Checkbox/Checkbox.vue')
  }
})
