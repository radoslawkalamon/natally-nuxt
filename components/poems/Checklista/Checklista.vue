<template>
  <div
    class="checklista"
    data-test="poems-checklista"
  >
    <ComponentsCheckbox
      checked="checked"
      class="checklista__checkbox"
      label="Byle by być"
      name="byle-by-byc"
    />
    <ComponentsCheckbox
      class="checklista__checkbox"
      label="Byle by zaliczyć"
      name="byle-by-zaliczyc"
    />
    <ComponentsCheckbox
      class="checklista__checkbox"
      label="Punkt po punkcie"
      name="punkt-po-punkcie"
    />
    <ComponentsCheckbox
      class="checklista__checkbox"
      label="Życie"
      name="zycie"
    />
    <p class="checklista__last-line">
      <span class="checklista__gradient">Życie w niebycie</span>
    </p>
  </div>
</template>

<script src="./Checklista.ts" lang="ts"></script>
<style src="./Checklista.css" scoped></style>
