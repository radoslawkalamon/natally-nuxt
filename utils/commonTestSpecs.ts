import Vue from 'vue'
import { mount } from '@vue/test-utils'
import { merge } from 'lodash'

export function shallRender (component: typeof Vue, dataTestID: string, defaultOptions?: object): void {
  test('shall render', (): void => {
    const wrapper = mount(component, defaultOptions)
    const isWrapperVisible = wrapper.find(`[data-test="${dataTestID}"]`).isVisible()
    expect(isWrapperVisible).toBeTruthy()
  })
}

export function shallHaveSlot (component: typeof Vue, slot: string, defaultOptions?: object): void {
  test(`shall have slot '${slot}'`, (): void => {
    const text = `This is slot '${slot}' test.`
    const wrapperOptions = merge(defaultOptions, {
      slots: {
        [slot]: text
      }
    })
    const wrapper = mount(component, wrapperOptions)
    expect(wrapper.html()).toContain(text)
  })
}

export function shallHaveStringProp (component: typeof Vue, propName: string, defaultOptions?: object): void {
  test(`shall have string prop '${propName}'`, (): void => {
    const text = `This is prop ${propName} test.`
    const wrapperOptions = merge(defaultOptions, {
      propsData: {
        [propName]: text
      }
    })
    const wrapper = mount(component, wrapperOptions)
    expect(wrapper.html()).toContain(text)
  })
}

export function shallHavePropInfluenceOverClassNames (
  component: typeof Vue,
  propName: string,
  propValue: unknown,
  classNameSuffix: string,
  defaultOptions?: object
) {
  test(`shall have ${typeof propValue} prop '${propName}' influence over class names`, () => {
    const wrapperOptions = merge(defaultOptions, {
      propsData: {
        [propName]: propValue
      }
    })
    const wrapper = mount(component, wrapperOptions)
    const isWrapperHaveClass = wrapper.classes().some(cls => cls.includes(classNameSuffix))
    expect(isWrapperHaveClass).toBeTruthy()
  })
}
