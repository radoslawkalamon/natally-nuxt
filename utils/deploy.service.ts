/* eslint-disable no-console */
import { Client, FileInfo, FTPResponse } from 'basic-ftp'
import { ProgressInfo } from 'basic-ftp/dist/ProgressTracker'
import capitalize from './formatters/capitalize'
import spacelize from './formatters/spacelize'

type DeployServiceConfig = {
  host: string;
  port: number;
  username: string;
  password: string;
  remoteCatalog: string;
  remoteIgnoreFile: string[];
  localCatalog: string;
}

type RemoveStrategies = {
  [key in FileInfo['type']]: (name: FileInfo['name']) => Promise<void | FTPResponse>
}

function decoratorLogCall () {
  return function (_target: any, propertyKey: string, descriptor: PropertyDescriptor) {
    const targetMethod = descriptor.value

    descriptor.value = function (...args: any[]) {
      const formattedName = spacelize(capitalize(propertyKey))
      console.log(`::: ${formattedName} :::`)
      return targetMethod.apply(this, args)
    }

    return descriptor
  }
}

class DeployService {
  readonly host: string;
  readonly port: number;
  readonly username: string;
  readonly password: string;
  readonly remoteCatalog: string;
  readonly remoteIgnoreFile: string[];
  readonly localCatalog: string;
  FTPClient = new Client()

  constructor (config: DeployServiceConfig) {
    this.host = config.host
    this.port = Number(config.port)
    this.username = config.username
    this.password = config.password
    this.remoteCatalog = config.remoteCatalog
    this.remoteIgnoreFile = config.remoteIgnoreFile
    this.localCatalog = config.localCatalog
  }

  public async start (): Promise<void> {
    try {
      await this.openConnection()
      await this.ensureRemoteCatalog()
      await this.openRemoteCatalog()
      await this.cleanRemoteCatalog()
      await this.uploadToRemoteCatalog()
    } catch (e) {
      console.error(e)
      throw e
    } finally {
      this.closeConnection()
    }
  }

  @decoratorLogCall()
  private async openConnection (): Promise<void> {
    await this.FTPClient.access({
      host: this.host,
      port: this.port,
      user: this.username,
      password: this.password,
      secure: false
    })
  }

  @decoratorLogCall()
  private async ensureRemoteCatalog (): Promise<void> {
    await this.FTPClient.ensureDir(this.remoteCatalog)
  }

  @decoratorLogCall()
  private async openRemoteCatalog (): Promise<void> {
    await this.FTPClient.cd(this.remoteCatalog)
  }

  @decoratorLogCall()
  private async cleanRemoteCatalog (): Promise<void> {
    const remoteFiles = await this.FTPClient.list()

    for (const file of remoteFiles) {
      if (this.remoteIgnoreFile.includes(file.name)) {
        continue
      }

      const removeStrategies: RemoveStrategies = {
        0: async () => await Promise.resolve(),
        1: async name => await this.FTPClient.remove(name),
        2: async name => await this.FTPClient.removeDir(name),
        3: async () => await Promise.resolve()
      }
      await removeStrategies[file.type](file.name)
    }
  }

  @decoratorLogCall()
  private async uploadToRemoteCatalog (): Promise<void> {
    this.FTPClient.trackProgress(this.handlerUploadProgress)
    await this.FTPClient.uploadFromDir(this.localCatalog)
    this.FTPClient.trackProgress()
  }

  private handlerUploadProgress (info: ProgressInfo): void {
    if (info.bytes === 0) {
      process.stdout.write(`Uploading file ${info.name}\n`)
    }
  }

  @decoratorLogCall()
  private closeConnection (): void {
    this.FTPClient.close()
  }
}

export default DeployService
