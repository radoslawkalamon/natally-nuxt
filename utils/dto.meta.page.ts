export type DTOMetaPageConstructor = {
  createdAt: string;
  description: string;
  imageOpenGraph?: string;
  path: string;
  title: string;
  updatedAt: string;
}

const getPath = (path: string) => {
  const pathWithoutIndex = path.split('/').filter(i => i !== 'index').join('/')
  return `${pathWithoutIndex}${pathWithoutIndex.at(-1) !== '/' ? '/' : ''}`
}

export class DTOMetaPage {
  readonly createdAt: Date;
  readonly description: string;
  readonly imageOpenGraph: string;
  readonly path: string;
  readonly title: string;
  readonly updatedAt: Date;

  constructor ({
    createdAt,
    description,
    imageOpenGraph = 'common.opengraph.png',
    title,
    updatedAt,
    path
  }: DTOMetaPageConstructor) {
    this.createdAt = new Date(createdAt)
    this.description = description
    this.imageOpenGraph = `/images/${imageOpenGraph}`
    this.path = getPath(path)
    this.title = title
    this.updatedAt = new Date(updatedAt)
  }
}
