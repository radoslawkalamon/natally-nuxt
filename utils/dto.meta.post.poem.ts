import { DTOMetaPage, DTOMetaPageConstructor } from '@/utils/dto.meta.page'

export type DTOMetaPostPoemConstructor = DTOMetaPageConstructor & {
  imageCover: string
  imageOpenGraph: string
}

export class DTOMetaPostPoem extends DTOMetaPage {
  readonly imageCover: string;
  readonly imageOpenGraph: string;

  constructor ({
    createdAt,
    description,
    imageCover,
    imageOpenGraph,
    path,
    title,
    updatedAt
  }: DTOMetaPostPoemConstructor) {
    super({
      createdAt,
      description,
      path,
      title,
      updatedAt
    })

    this.imageCover = `${this.imagePath}${imageCover}`
    this.imageOpenGraph = `${this.imagePath}${imageOpenGraph}`
  }

  get imagePath (): string {
    return '/images/poezja-314/'
  }
}
