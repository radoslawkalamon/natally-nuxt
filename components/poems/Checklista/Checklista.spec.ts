import Checklista from './Checklista.vue'
import { shallRender } from '@/utils/commonTestSpecs'

describe('Poems / Checklista', () => {
  const defaultOptions = {
    stubs: {
      ComponentsCheckbox: {
        template: '<input data-stub="components-checkbox" type="checkbox" />'
      }
    }
  }

  shallRender(Checklista, defaultOptions)
})
