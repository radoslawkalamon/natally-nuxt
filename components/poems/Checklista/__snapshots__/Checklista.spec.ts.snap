// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poems / Checklista Unit shall pass unit sanity test 1`] = `
"<div data-test=\\"poems-checklista\\" class=\\"checklista\\"><label for=\\"byle-by-byc\\" class=\\"checklista__checkbox\\"><input id=\\"byle-by-byc\\" type=\\"checkbox\\" name=\\"byle-by-byc\\" class=\\"checklista__input\\"> <span class=\\"checklista__label\\">Byle by być</span></label><label for=\\"byle-by-zaliczyc\\" class=\\"checklista__checkbox\\"><input id=\\"byle-by-zaliczyc\\" type=\\"checkbox\\" name=\\"byle-by-zaliczyc\\" class=\\"checklista__input\\"> <span class=\\"checklista__label\\">Byle by zaliczyć</span></label><label for=\\"punkt-po-punkcie\\" class=\\"checklista__checkbox\\"><input id=\\"punkt-po-punkcie\\" type=\\"checkbox\\" name=\\"punkt-po-punkcie\\" class=\\"checklista__input\\"> <span class=\\"checklista__label\\">Punkt po punkcie</span></label><label for=\\"zycie\\" class=\\"checklista__checkbox\\"><input id=\\"zycie\\" type=\\"checkbox\\" name=\\"zycie\\" class=\\"checklista__input\\"> <span class=\\"checklista__label\\">Życie</span></label>
  <p class=\\"checklista__last-line\\"><span class=\\"checklista__gradient\\">Życie w niebycie</span></p>
</div>"
`;
